sql/hive/src/test/scala/org/apache/spark/sql/hive/OptimizeHiveMetadataOnlyQuerySuite.scala

```diff
@@ -65,11 +65,11 @@ class OptimizeHiveMetadataOnlyQuerySuite extends QueryTest with TestHiveSingleto
 
       // verify the matching partitions
       val partitions = spark.internalCreateDataFrame(Distinct(Filter(($"x" < 5).expr,
-        Project(Seq(($"part" + 1).as("x").expr.asInstanceOf[NamedExpression]),
+        Project(Seq(($"part" * 1).as("x").expr.asInstanceOf[NamedExpression]),
```

Comment:
why do we need this?

Reply:
Because with this optimizer in this PR, `part + 1  < 5` will be optimized to 'part < 4' , where `part` is a partition column, so it only need to fetch 4 partitions instead of 11, so the last assert `assert(HiveCatalogMetrics.METRIC_PARTITIONS_FETCHED.getCount - startCount == 11)` will fail.   
From the comments in this test, it wants to test the case where `verify that the partition predicate was not pushed down to the metastore`, so I changed to `part * 1`, which will not be optimized.

Reply:
How about using the `spark.sql.optimizer.excludedRules` config instead of changing the existing tests?

Reply:
> How about using the `spark.sql.optimizer.excludedRules` config instead of changing the existing tests?

Thanks for your advice. I've make change accordingly.